src/oscar/apps/customer/utils.py

```diff
@@ -75,7 +75,7 @@ def create_customer_email(self, user, messages, email):
         Create Email instance in database for logging purposes.
         """
         # Is user is signed in, record the event for audit
-        if email and user_is_authenticated(user):
+        if email and user.is_authenticated():
```

Comment:
This is invalid in Django 2 (`is_authenticated` is a property, not a method). I'd suggest you leave this file unchanged for now, and we update everything at once in #2496.